sdk/samples/iot/sample_pnp.h

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef SAMPLE_PNP_H
+#define SAMPLE_PNP_H
+
+#include <stdint.h>
+
+#include <azure/core/az_json.h>
+#include <azure/core/az_result.h>
+#include <azure/core/az_span.h>
+#include <azure/iot/az_iot_hub_client.h>
+
+#define PNP_STATUS_SUCCESS 200
+#define PNP_STATUS_BAD_FORMAT 400
+#define PNP_STATUS_NOT_FOUND 404
+#define PNP_STATUS_INTERNAL_ERROR 500
+
+typedef void (*pnp_property_callback)(
+    az_span component_name,
+    az_span property_name,
+    az_json_token* property_value,
```

Comment:
>az_json_token [](start = 4, length = 13)

How about sharing the copy of reader with caller, rather then token. In this way if callback can also parse complex objects.

Reply:
For the purposes of, I'd say that might bring in more complex situations than we're looking for. If you disagree please let me know.

Reply:
updated to pass the json reader